src/Symfony/Component/Security/Core/Authentication/Token/RememberMeToken.php

```diff
@@ -52,14 +50,20 @@ public function getFirewallName(): string
         return $this->firewallName;
     }
 
+    /**
+     * @deprecated since Symfony 7.2
+     */
     public function getSecret(): string
     {
-        return $this->secret;
+        trigger_deprecation('symfony/security-core', '7.2', 'The "%s()" method is deprecated.', __METHOD__);
+
+        return $this->secret ??= base64_encode(random_bytes(8));
     }
 
     public function __serialize(): array
     {
-        return [$this->secret, $this->firewallName, parent::__serialize()];
+        // $this->firewallName should be kept at index 1 for compatibility with payloads generated before Symfony 8
+        return [$this->secret ?? null, $this->firewallName, parent::__serialize()];
```

Comment:
Why are we guarding `$this->secret` with a null-coalescing operator? The property should always exist, shouldn't it?

```suggestion
        return [$this->secret, $this->firewallName, parent::__serialize()];
```

Reply:
indeed, updated